src/org/openstreetmap/josm/gui/dialogs/properties/PropertiesDialog.java

```diff
@@ -635,7 +671,37 @@ public void selectionChanged(SelectionChangeEvent event) {
 
         // Ignore parameter as we do not want to operate always on real selection here, especially in draw mode
         Collection<? extends IPrimitive> newSel = OsmDataManager.getInstance().getInProgressISelection();
-        int newSelSize = newSel.size();
+
+        updateUi(newSel);
+    }
+
+    @Override
+    public void mouseMoved(MouseEvent e) {
+        if (!OsmDataManager.getInstance().getInProgressISelection().isEmpty())
+            return;
+
+        MapView mv = MainApplication.getMap().mapView;
+        Optional<OsmPrimitive> p = Optional.ofNullable(
+                mv.getNearestNodeOrWay(e.getPoint(), mv.isSelectablePredicate, true));
+
+        List<OsmPrimitive> hovered = new ArrayList<>();
+        p.ifPresent(hovered::add);
+        updateUi(hovered);
+    }
+
+    @Override
+    public void mouseDragged(MouseEvent e) {
+        // Do nothing
+    }
```

Comment:
I'm really not a fan of this, but we don't have a better option in core right now.

I'd like to have a `HighlightManager` or something which passes highlight/hover information around, but we don't have that yet. :(

Reply:
Yeah same, it's not the prettiest piece of code but the only real option at the moment. I might look into adding a `HoverListener` to the map view that client code can use to be notified whenever the object under the mouse pointer changes. That could be reused in all sorts of places.